java/src/jmri/server/json/memory/JsonMemoryHttpService.java

```diff
@@ -32,10 +45,31 @@ public ObjectNode doGet(Memory memory, String name, String type, Locale locale,
         ObjectNode root = this.getNamedBean(memory, name, type, locale, id);
         ObjectNode data = root.with(DATA);
         if (memory != null) {
-            if (memory.getValue() == null) {
+            Object val = memory.getValue();
+            if (val == null) {
                 data.putNull(VALUE);
             } else {
-                data.put(VALUE, memory.getValue().toString());
+                //set memory value based on data type
+                if (val instanceof String) {
+                    data.put(VALUE, (String) val);
+                } else if (val instanceof javax.swing.ImageIcon) {
+                    log.warn("ImageIcon not yet supported");
+                    data.putNull(VALUE);
+                } else if (val instanceof Number) {
+                    data.put(VALUE, val.toString());
+                } else if (val instanceof jmri.IdTag){
+                    // most IdTags are Reportable objects, so 
+                    // this needs to be before Reportable
+                    ObjectNode idTagValue = idTagService.doGet((jmri.IdTag)val, name, IDTAG, locale, id);
+                    data.set(VALUE, idTagValue);
+                } else if (val instanceof Reportable) {
+                    ObjectNode reporterValue = reporterService.doGet((jmri.Reporter)val, name, REPORTER, locale, id);
+                    data.set(VALUE, reporterValue);
+                } else {
+                    log.warn("can't return current value of memory '{}', val='{}' of Class '{}'", 
```

Comment:
This catch-all should put the result of `val.toString()` into the Memory instead logging a warning and returning a null value.